src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -17,7 +18,10 @@ public User login(String login, String password)
             throws AuthenticationException {
         Optional<User> userFromDB = userService.findByLogin(login);
 
-        if (userFromDB.isPresent() && userFromDB.get().getPassword().equals(password)) {
+        byte[] salt = userFromDB.get().getSalt();
```

Comment:
we are calling `get()` on Optional here
but we haven't checked yet if the value is present